### DISCUSSION TITLE
SPARQL (GraphDB) and Solr Integrations

### DISCUSSION THREAD

**Discussion by malajvan** [DDMAL/linkedmusic-datalake#4] [Ideas]
As Ich mentioned, in the spirit of "making search smarter," we look into utilizing Solr's powerful backend with SPARQL queries, combining full-text search with graph-based querying. 
Currently, I'm looking at 3 options: [SolRDF](https://github.com/spaziocodice/SolRDF) and [Solr GraphDB connector](https://graphdb.ontotext.com/documentation/10.0/solr-graphdb-connector.html), and develop our own extension. I'll discuss below why I think with these options, developing our own connector would be the most realistic, however, I look forward to others' ideas and alternative solutions.

1. [SolRDF](https://github.com/spaziocodice/SolRDF) 
Solr extensions for managing RDF data. SolRDF is also fully compliant with SPARQL 1.1 endpoint. It is however pretty old and the last update was 8 years ago on the repo, and supports Solr 5.x (The latest Solr version is 9.3.0)
Because of these reasons, I don't think we should implement SolRDF for LinkedMusic, but it's still worth looking at for references in case we want to write our own extension/package.

2. [Solr GraphDB connector](https://graphdb.ontotext.com/documentation/10.0/solr-graphdb-connector.html)
GraphDB's official connector to Solr, which provides extensive support for querying Solr full-text and faceted search, entity synchronization between Solr index and GraphDB data. 
This connector however is only available in the GraphDB Enterprise license, which requires a (probably expensive) subscription which is not realistic for our context.

My current idea for moving forward is leaning towards either writing our own connectors or implementing SolRDF but updated to be compatible with the latest versions of Solr. This might be a discussion more suitable later on, but I think it's beneficial to look at these options as we figure out the pipelines.

**Comment by fujinaga** [comment]
Can you also look at Virtuoso and Apache Jena?

On Aug 8, 2023, at 11:58 AM, Van Pham ***@***.***> wrote:



As Ich mentioned, in the spirit of "making search smarter," we look into utilizing Solr's powerful backend with SPARQL queries, combining full-text search with graph-based querying.
Currently, I'm looking at 3 options: SolRDF<https://github.com/spaziocodice/SolRDF> and Solr GraphDB connector<https://graphdb.ontotext.com/documentation/10.0/solr-graphdb-connector.html>, and develop our own extension. I'll discuss below why I think with these options, developing our own connector would be the most realistic, however, I look forward to others' ideas and alternative solutions.

  1.  SolRDF<https://github.com/spaziocodice/SolRDF>
Solr extensions for managing RDF data. SolRDF is also fully compliant with SPARQL 1.1 endpoint. It is however pretty old and the last update was 8 years ago on the repo, and supports Solr 5.x (The latest Solr version is 9.3.0)
Because of these reasons, I don't think we should implement SolRDF for LinkedMusic, but it's still worth looking at for references in case we want to write our own extension/package.

  2.  Solr GraphDB connector<https://graphdb.ontotext.com/documentation/10.0/solr-graphdb-connector.html>
GraphDB's official connector to Solr, which provides extensive support for querying Solr full-text and faceted search, entity synchronization between Solr index and GraphDB data.
This connector however is only available in the GraphDB Enterprise license, which requires a (probably expensive) subscription which is not realistic for our context.

My current idea for moving forward is leaning towards either writing our own connectors or implementing SolRDF but updated to be compatible with the latest versions of Solr. This might be a discussion more suitable later on, but I think it's beneficial to look at these options as we figure out the pipelines.
